### PR TITLE
Fixes axes reordering for data that has no channel axis

### LINKED ISSUE #1845: Axis reordering with singleton axes broken
I have a dataset in an h5 file, automatically recognized to the order of zyxc, with dimensions (512, 513, 514, 1). When I try to re-order to xyzc, it complains that `Exception: Your dataset's provided axistags (xyzc) do not have the correct dimensionality for your dataset, which has 3 dimensions`. If I try with just xyz, it complains that `can not set axes, wrong number`.

### LINKED ISSUE #1848: Batch processing on image stacks
I want to run pixel classification of image stacks with multiple channels. Training can be done by changing the axes from 'zyx' to 'cyx'. however this can't seem to be done when loading the same multichannel tiffs in to the batch processer.

Version: Ilastik 1.3.0 for OsX

### PR DESCRIPTION
**Pull request by k-dominik**
I would recommend merging this one in favor of changing the hdf5 reader (for now). The code change is way more minimal and I also found, that the hdf5 reader is not the only one allowing for no channel axis.

fixes #1845

ToDo:

* [x] check whether this fixes #1848

checks against the `_NontransposedImage` axes (no channel added) whether a channel
has has been provided by the respective reader instance, or was later added.

### CODE REVIEW

**Comment by k-dominik** [comment]
heyo @FynnBe, do you mind taking a look?

File: ilastik/applets/dataSelection/datasetInfoEditorWidget.py [hunk 1]
```diff
@@ -442,13 +442,23 @@ def _applyAxesToTempOps(self):
...
             numaxes = len(original_shape)
         else:
             numaxes = len(shape)
 
+        # This portion was added in order to handle the OpDataSelection adding
+        # a channel axis when encountering data without one.
+        # check if channel was added and not present in original:
+        axistags = firstOp._NonTransposedImage.meta.original_axistags
+        if axistags is None:
+            axistags = firstOp._NonTransposedImage.meta.axistags
```
Comment (FynnBe) [inline ilastik/applets/dataSelection/datasetInfoEditorWidget.py]: How about using metaDict.getOriginalAxisKeys() here?

**Comment by k-dominik** [comment]
```
Encountered 3 failures in the following files:
./test_applets/featureSelection/testFeatureSelectionSerializer.py
./test_applets/pixelClassification/testPixelClassificationGui.py
./test_applets/pixelClassification/testPixelClassificationMultiImageGui.py
```
these look like legit failures that were not captured previously, due to the dependency problem

**Comment by k-dominik** [comment]
I just gave it a try, and i get the following error after adding a file, changing the axes, clicking on the feature selection applet and hitting the "select features" button:

```pytb
ERROR 2018-10-10 17:05:20,656 excepthooks 6590 140397503596352 Unhandled exception in thread: 'MainThread'
ERROR 2018-10-10 17:05:20,656 excepthooks 6590 140397503596352 Traceback (most recent call last):
  File "/.../sources/ilastik-meta/ilastik/ilastik/applets/featureSelection/featureSelectionGui.py", line 305, in onFeatureButtonClicked
    self.topLevelOperatorView.SelectionMatrix.setValue(self.topLevelOperatorView.MinimalFeatures)
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 118, in call_in_setup_context
    return func(self, *args, **kwargs)
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1232, in setValue
    self._changed()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1433, in _changed
    c._changed()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1439, in _changed
    self._configureOperator(self)
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1453, in _configureOperator
    self.operator._setupOutputs()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/operator.py", line 472, in _setupOutputs
    oslot._setReady()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1352, in _setReady
    self._changed()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1433, in _changed
    c._changed()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1439, in _changed
    self._configureOperator(self)
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/slot.py", line 1453, in _configureOperator
    self.operator._setupOutputs()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/operator.py", line 455, in _setupOutputs
    self.setupOutputs()
  File "/.../miniconda3/envs/idev/ilastik-meta/lazyflow/lazyflow/operators/opSlicedBlockedArrayCache.py", line 103, in setupOutputs
    if len(blockshape) != len(self.Input.meta.shape):
TypeError: object of type 'NoneType' has no len()
```

**Comment by k-dominik** [comment]
Thank you very much @FynnBe for digging into it. I think this is in a very good state to be merged (so that's what I'm going to do)!

**Pull request merged**